225 add an about page

shrprabh opened the pull request:
This pull request includes several changes to the `src/components`, `src/layout`, and `src/routes` directories to add a new `About` page and improve type safety in the `CopyToClipboardButton` component. The most important changes include adding the `About` page component, updating the main layout to include a link to the `About` page, and adding type definitions to the `CopyToClipboardButton` component.

### New `About` Page:

* [`src/components/About.tsx`](diffhunk://#diff-8153b95ba37a69296bb9a433eabaafdbcec247f70529d101574ef340b2f032fbR1-R201): Added a new `About` page component that includes detailed information about BilboMD, its features, and pipeline options.

### Main Layout Updates:

* [`src/layout/MainLayout/index.tsx`](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcR129-R135): Added an `About` menu item to the main layout navigation, allowing users to access the new `About` page.
* [`src/layout/MainLayout/index.tsx`](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcL17-R18): Corrected the import statement for `InfoOutlined` icon.

### Route Updates:

* [`src/routes/MainRoutes.tsx`](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R89-R92): Added a new route for the `About` page under the `dashboard` path.
* [`src/routes/MainRoutes.tsx`](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R18-R19): Imported the `About` component to be used in the new route.

### Type Safety Improvements:

* [`src/components/Common/CopyToClipboardButton.tsx`](diffhunk://#diff-8b590fe07d9901337e5185b86ecf10cc5993670fa368c9876d4740189114d58fL5-R9): Added TypeScript interface for `CopyToClipboardButton` component props to improve type safety.
![image](https://github.com/user-attachments/assets/36eeea82-f39f-4538-94c7-3612dd8e32ac)

shrprabh commented:
@dsclassen  In an attempt to repurpose the home component and direct it as an about component, I discovered that the home component was redirecting to the welcome component after token verification. To enhance clarity, I introduced a new component. If necessary, we can relocate this component to a different directory.

shrprabh commented:
there are few changes i need to add to include current features will be pushing the changes.

dsclassen commented:
I do remember this being an issue... unauthenticated users get the `src/components/Home.tsx` component. Authenticated users get bounced to `src/features/auth/Welcome.tsx`.

I think it makes good sense to have a dedicated `About.tsx` component and it can certainly reuse stuff from `Home`

dsclassen commented:
I would like to not add new code to the `components` folder.  In fact I would like to slowly move everything out of that directory into `features`. Perhaps a new directory `features/about` ?

shrprabh commented:
Thank you for your feedback. I will relocate the “about” to **features directory**. Additionally, I have created four reusable components by creating a shared directory under components directory to enhance reusability and facilitate unit testing. I will be moving that as well to the “features” directory. These components can be utilized in the “home” component as well. **If this is acceptable,** I can incorporate them into the “home” component immediately. I have not utilized the shared components in the “home” component, but I have used them in the “about” component.


### New Components [Shared Components ]:
![image](https://github.com/user-attachments/assets/74a1ae91-62c8-4349-a29b-9314c9b22189)

* [`src/components/About.tsx`](diffhunk://#diff-8153b95ba37a69296bb9a433eabaafdbcec247f70529d101574ef340b2f032fbR1-R221): Introduced a new `About` component that provides detailed information about BilboMD, including features, pipeline options, and additional resources.
* [`src/components/Shared/AdditionalInfo.tsx`](diffhunk://#diff-d2dc169846e8311e8d0f0eff357961eaa780bf94d6b51457928ae130a4ed8cf2R1-R15): Added a new `AdditionalInfo` component to display information about ongoing development and provide a link to the GitHub repository for feedback.
* [`src/components/Shared/FeaturesList.tsx`](diffhunk://#diff-0524be68894438ef4c45cbbcf163845c4fd1cf87f3616409ad954b7bdaa5ec10R1-R16): Created a `FeaturesList` component to list the current features of BilboMD.
* [`src/components/Shared/Introduction.tsx`](diffhunk://#diff-0341b67e787baf5c052975d82a7d995f0de0bb466dbc413ab788f155cb33c8caR1-R19): Added an `Introduction` component to provide an introductory section for the `About` page.
* [`src/components/Shared/PipelineOptions.tsx`](diffhunk://#diff-25022d047ec93576f33068044d5079741cf224a816fed830c99de329cb2182b1R1-R69): Introduced a `PipelineOptions` component to display different pipeline options available in BilboMD.

### TypeScript Enhancements:

* [`src/components/Common/CopyToClipboardButton.tsx`](diffhunk://#diff-8b590fe07d9901337e5185b86ecf10cc5993670fa368c9876d4740189114d58fL5-R9): Updated the `CopyToClipboardButton` component to use TypeScript by defining `CopyToClipboardButtonProps` interface.

### Routing and Layout Updates:

* [`src/layout/MainLayout/index.tsx`](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcL17-R18): Updated the layout to include a new "About" menu item that navigates to the `About` page. [[1]](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcL17-R18) [[2]](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcR129-R135)
* [`src/routes/MainRoutes.tsx`](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R18-R19): Added routing for the new `About` page under the dashboard route. [[1]](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R18-R19) [[2]](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R89-R92)
![image](https://github.com/user-attachments/assets/dc1e073c-861e-43d2-9c7b-0601932b21bb)

dsclassen commented:
yes, please complete your refactor by incorporating the new components into the `Home` component.

shrprabh commented:
Hello @dsclassen, I’ve refactored the code and moved the about component to the features directory. I’ve verified the code in light and dark modes, including login verification, loaders, and authentication. Please review the code and let me know if it’s simple and good. If not, suggest areas for improvement. Currently, the home component is in the components directory, while about and shared components are in features. I can move home to features if needed.

dsclassen commented on src/features/about/About.tsx:
```diff
@@ -0,0 +1,221 @@
+        title: 'BilboMD Classic w/PDB',
+        description: 'Bring your own starting PDB model and constraints.',
+        imagePath: {
+          light: '/images/bilbomd-classic-pdb-schematic.png',
+          dark: '/images/bilbomd-classic-pdb-schematic.png'
```
your light and dark image paths are identical

shrprabh replied:
Updated with proper images for all the four pipelines.

dsclassen commented on src/features/about/About.tsx:
```diff
@@ -0,0 +1,221 @@
+import PipelineOptions from '../shared/PipelineOptions'
+import AdditionalInfo from '../shared/AdditionalInfo'
+
+const About = ({ title = 'About BilboMD' }) => {
+  useTitle(title)
```
I don't think that `title` needs to be a prop

```javascript
const About = () => {
useTitle("BilboMD: About")
```

should suffice

dsclassen replied:
I see what you're doing here. Please disregard my comment and leave as-is. Maybe change default value to:

```
BilboMD: About
```

shrprabh replied:
Sure, Done

dsclassen commented on src/features/about/About.tsx:
```diff
@@ -0,0 +1,221 @@
+      </>,
+      'You can see the status and history of your BilboMD jobs.',
+      'Provides a download link to retrieve the <code>results.tar.gz</code> file.',
+      'Currently we store BilboMD job results for 60 days, so be sure to download your results if they are important to you.'
```
add <b></b> tags to BilboMD please

shrprabh replied:
Done

dsclassen commented on src/features/shared/AdditionalInfo.tsx:
I don't think we will ever reuse this component with different values for `githubLink`
Maybe just hardcode the repo URL

shrprabh replied:
currently we are using it in Home.tsx and About.tsx. is it fine to hardcode it in both or shall i leave as it is         <AdditionalInfo githubLink='https://github.com/bl1231/bilbomd-ui' />

dsclassen replied:
What is better.. hardcoding in one place or in 2 places?

shrprabh replied:
@dsclassen Perfect! I initially misunderstood and thought I needed to remove the additionalInfo.tsx file. However, we only need to hardcode it in additionalInfo.tsx, and we can reuse it without passing props. That’s great!. Added the changes.

dsclassen commented on src/routes/MainRoutes.tsx:
```diff
@@ -84,6 +86,10 @@ const ProtectedMainRoutes = {
                   path: 'dashboard',
                   children: [
                     { path: '', element: <Welcome /> },
+                    {
+                      path: 'about',
+                      element: <About title='About BilboMD' />
+                    },
                     {
```
I see now why you were passing `title` into `About`.
I kind of like this pattern. I may implement this throughout the entire application.
Please keep it for now.

shrprabh replied:
Corrected

dsclassen reviewed (commented):
please see comments

shrprabh commented on src/components/Home.tsx:
I have added added all the corrections.

shrprabh commented on src/features/about/About.tsx:
```diff
@@ -0,0 +1,221 @@
+        </ul>
+      </>,
+      'You can see the status and history of your BilboMD jobs.',
+      'Provides a download link to retrieve the <code>results.tar.gz</code> file.',
```
corrected by using the react fragments.

dsclassen commented on src/layout/MainLayout/index.tsx:
```diff
@@ -126,6 +126,13 @@ export default function ClippedDrawer() {
+      icon: <InfoOutlined />,
+      path: '/dashboard/about',
+      onclick: () => navigate('/dashboard/about'),
+      roles: ['user', 'manager', 'admin']
```
Our filtering logic is a bit wonky here. I can't see the `About` menu item as a user with role = User

Please change to:

```javascript
    {
      text: 'About',
      icon: <InfoOutlined />,
      path: '/dashboard/about',
      onclick: () => navigate('/dashboard/about'),
      roles: ['user']
    }
```

and while you're at it you can change all the ones that currently say:

```javascript
roles: ['user', 'manager']
```

to:

```javascript
roles: ['user']
```

shrprabh replied:
It implies that the manager and admins are also default users.

dsclassen reviewed (changes requested):
Just change the menu item filtering stuff and I think we're good to merge.
